actionview/test/template/template_test.rb

```diff
@@ -314,4 +326,111 @@ def test_template_inspect
     @template = new_template("hello")
     assert_equal "#<ActionView::Template hello template locals=[]>", @template.inspect
   end
+
+  def test_template_translate_location
+    highlight = "nomethoderror"
+    source = "<%= nomethoderror %>"
+    compiled = "'.freeze; @output_buffer.append=  nomethoderror ; @output_buffer.safe_append='\n"
+
+    backtrace_location = Data.define(:lineno).new(lineno: 1)
+    spot = spot_highlight(compiled, highlight)
+    expected = spot_highlight(source, highlight, snippet: compiled)
+
+    assert_equal expected, new_template(source).translate_location(backtrace_location, spot)
+  end
+
+  def test_template_translate_location_lineno_offset
+    highlight = "nomethoderror"
+    source = "<%= nomethoderror %>"
+    compiled = "\n'.freeze; @output_buffer.append=  nomethoderror ; @output_buffer.safe_append='\n"
+
+    backtrace_location = Data.define(:lineno).new(lineno: 1)
+    spot = spot_highlight(compiled, highlight, first_lineno: 2, last_lineno: 2)
+    expected = spot_highlight(source, highlight, snippet: compiled)
+
+    assert_equal expected, new_template(source).translate_location(backtrace_location, spot)
+  end
+
+  def test_template_translate_location_with_multibye_string_before_highlight
```

Comment:
This test is failing on CI:

```
Failure:
TestERBTemplate#test_template_translate_location_with_multibye_string_before_highlight [test/template/template_test.rb:363]:
--- expected
+++ actual
@@ -1,2 +1,3 @@
-{:first_column=>5, :last_column=>18, :snippet=>"¥'.freeze; @output_buffer.append=  nomethoderror ; @output_buffer.safe_append='
-", :first_lineno=>1, :last_lineno=>1, :script_lines=>["¥<%= nomethoderror %>"]}
+{:first_column=>35, :last_column=>48, :snippet=>"¥'.freeze; @output_buffer.append=  nomethoderror ; @output_buffer.safe_append='
+", :first_lineno=>1, :last_lineno=>1, :script_lines=>["¥'.freeze; @output_buffer.append=  nomethoderror ; @output_buffer.safe_append='
+"]}
```

Reply:
I noticed this last night too. 😬 It _was_ passing so I'll figure out what I changed.

The only failure is the multi-byte test so it's probably some byte/char inconsistency. I'll update when it's fixed.

Reply:
There was still an unrelated failure. I re-pushed after clarifying the CHANGELOG entry to force a rebuild, but I'll just leave it alone if it fails obtaining a connection again and you can retrigger the build.